appinventor/appengine/src/com/google/appinventor/client/editor/simple/ComponentDatabase.java

```diff
@@ -331,9 +331,15 @@ private boolean initComponent(JSONObject componentNode) {
   private void findComponentProperties(ComponentDefinition component, JSONArray propertiesArray) {
     for (JSONValue propertyValue : propertiesArray.getElements()) {
       Map<String, JSONValue> properties = propertyValue.asObject().getProperties();
+
+      List<String> editorArgsList = new ArrayList<String>();
+      for (JSONValue val : properties.get("editorArgs").asArray().getElements())
```

Comment:
This line causes a fault at runtime when loading a project. Basically editorArgs do not exist on existing components and are not present in existing projects. You should either deal with the case where editorArgs doesn't exist, or write a project upgrader that makes sure it does!

Reply:
Sorry, I missed this point. Can you give me some guidance on writing a project upgrader?

Reply:
Currently I cannot reproduce the bug. I creates some .aia files from MIT official server. My server ([link](http://wtf.csie.org:8888/)) imports these files without error.

Here are the configs for my build.
* ArchLinux with OpenJDK 7 as default.
* Compile with `ant clean && ant -Ddebug=true`.
* Start the server with `dev_appserver.sh --jvm_flag=-Xdebug --jvm_flag=-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=55555 --address=0.0.0.0 --port=8888 appengine/build/war` and `ant RunLocalBuildServer`.